swarm/storage/resource.go

```diff
@@ -604,15 +604,35 @@ func (self *ResourceHandler) parseUpdate(chunkdata []byte) (*Signature, uint32,
 	headerlength := binary.LittleEndian.Uint16(chunkdata[cursor : cursor+2])
 	cursor += 2
 	datalength := binary.LittleEndian.Uint16(chunkdata[cursor : cursor+2])
-	exclsignlength := int(headerlength + datalength + 4)
-	if exclsignlength > len(chunkdata) || exclsignlength < 14 {
+	cursor += 2
+	var exclsignlength int
+	// we need extra magic if it's a multihash
+	// TODO: merge with isMultihash code
+	if datalength == 0 {
+		uvarintbuf := bytes.NewBuffer(chunkdata[headerlength+4:])
+		r, err := binary.ReadUvarint(uvarintbuf)
+		if err != nil {
+			return nil, 0, 0, "", nil, false, NewResourceError(ErrCorruptData, fmt.Sprintf("corrupt multihash, hash id varint could not be read: %v", err))
```

Comment:
Corrupt data sounds bad - maybe add a warning or error if we expect NOT to get in this block often?

Reply:
Should bubble up and handled by the caller? The `ErrCorruptData` distinguishes the error type, and theoretically this method could be used to check integrity of arbitrary data.